### ISSUE TITLE
Shellcheck failure on repeatable argument with allowed values

### ISSUE THREAD

**Issue by jhakonen** [DannyBen/bashly#534]
### Bashly Version

Latest Ruby Gem

### Description

Shellcheck fails when a command argument is repeatable and has list of allowed values.

Tested with bashly 1.2.0 and shellcheck 0.10.0.

### Contents of bashly.yml

```yaml
name: cli
commands:
- name: download
  args:
  - name: target
    allowed: [foo]
    repeatable: true
```

### Reproduction Steps

1. Run `bashly generate`
2. Run `shellcheck cli`

### Actual Behavior

```
$ shellcheck cli
In cli line 283:
  for i in "${input_array[@]}"; do
            ^---------------^ SC2154 (warning): input_array is referenced but not assigned.

For more information:
  https://www.shellcheck.net/wiki/SC2154 -- input_array is referenced but not...
```

### Expected Behavior

Shellcheck finishes with success.

**Comment by DannyBen** [comment]
Confirmed.

Additionally, this should have been caught by this CI test:

https://github.com/DannyBen/bashly/blob/868b2d7ee271723a53f1e6812df3e4877c2550d5/.github/workflows/test.yml#L78-L85

I will look into both issues.

**Comment by DannyBen** [comment]
This is fixed in #535.
Initially I tried to get rid of the `eval` there, but it proved to be more challenging to remove it and still not break anything.

More details in the PR if you are interested.

Thanks for taking the time to report, release will probably come next weekend.

**Comment by DannyBen** [comment]
Fixed in 1.2.1
